Forge.TreeWalker/src/ITreeWalkerCallbacks.cs

```diff
@@ -28,7 +26,9 @@ public interface ITreeWalkerCallbacks
         /// <param name="properties">The additional properties for this node.</param>
         /// <param name="userContext">The dynamic user-defined context object.</param>
         /// <param name="token">The cancellation token.</param>
-        Task BeforeVisitNode(Guid sessionId, string treeNodeKey, dynamic properties, object userContext, CancellationToken token);
+        /// <param name="treeName">The name of the ForgeTree in the JsonSchema.</param>
+        /// <param name="rootSessionId">The unique identifier for the root/parent tree walking session.</param>
+        Task BeforeVisitNode(Guid sessionId, string treeNodeKey, dynamic properties, object userContext, CancellationToken token, string treeName, Guid rootSessionId);
```

Comment:
https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1068?view=vs-2019
CancellationToken should be the last parameter. This comment applies to all occurences of this.

Can we enable FxCop for Forge Projects?

Reply:
Moved token to the end. I was initially thinking to not break backwards compatibility by changing the parameters order. But I'm breaking backwards compatibility anyways since these are new required parameters, so no harm in moving the token param to the end. :)